packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -886,3 +886,392 @@ RAW_TOOLS_DIR: ${Env:RAW_TOOLS_DIR}
+
+        } else {
+            $validatedData = $data
+        }
+
+        # check if path exists. If not, create the path for the registry value
+        if (!(Test-Path -Path $path)) {
+            # Create the registry key
+            New-Item -Path $path -Force | Out-Null
+            VM-Write-Log "INFO" "`t[+] Registry key created: $path"
+        } else {
+            VM-Write-Log "WARN" "`t[+] Registry key already exists: $path"
+        }
+
+        Set-ItemProperty -Path $path -Name $value -Value $validatedData -Type $type -Force | Out-Null
+        VM-Write-Log "INFO" "[+] $name has been successful"
+    } catch {
+        VM-Write-Log "ERROR" "Failed to update the registry value. Error: $_"
+    }
+}
+
+function VM-Remove-Path {
+# Function for removing Paths/Programs
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateSet("file", "dir")]
+        [string]$type,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$path
+    )
+
+    try {
+        if ($type -eq "file") {
+            if (Test-Path -Path $path -PathType Leaf) {
+                Remove-Item -Path $path -Force
+                VM-Write-Log "INFO" "[+] $name has been successfully removed."
+            } else {
+                VM-Write-Log "WARN" "[+] $path does not exist."
+            }
+        } elseif ($type -eq "dir") {
+            if (Test-Path -Path $path -PathType Container) {
+                Remove-Item -Path $path -Recurse -Force
+                VM-Write-Log "INFO" "[+] $name has been successfully removed."
+            } else {
+                VM-Write-Log "WARN" "[+] $path does not exist."
+            }
+        }
+    } catch {
+        VM-Write-Log "ERROR" "An error occurred while removing the $type $path. Error: $_"
+    }
+}
+
+function VM-Execute-Custom-Command{
+# Function for removing items in need of custom code.
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string[]]$cmds
+    )
+
+    try {
+        VM-Write-Log "INFO" "Executing commands for '$name':"
+        foreach ($cmd in $cmds) {
+            Start-Process powershell -ArgumentList "-WindowStyle","Hidden","-Command",$cmd -Wait
+        }
+        VM-Write-Log "INFO" "[+] All commands for '$name' have been executed successfully."
+    } catch {
+        VM-Write-Log "ERROR" "An error occurred while executing commands for '$name'. Error: $_"
+    }
+}
+
+function VM-Configure-Prompts {
+    # $Env:MandiantVM must be set in the install script
+    try {
+        # Set PowerShell prompt
+        $psprompt = @"
+        function prompt {
+            Write-Host (`$Env:MandiantVM + " " + `$(Get-Date)) -ForegroundColor Green
```

Comment:
Let's finalize a new variable name for `Env:MandiantVM` and update it's references everywhere 😄 How about `Env:VMname`?